website/static/js/waterbutler.js

```diff
@@ -24,65 +21,58 @@ function getDefaultOptions(path, provider) {
 
 function buildUrl(suffix, path, provider, nid, options) {
     path = path || '/';
-    var baseUrl = window.contextVars.waterbutlerURL + suffix;
-    return baseUrl + $.param($.extend(getDefaultOptions(path, provider), {nid: nid}, options));
-}
-
-var buildCrudUrl = buildUrl.bind(this, 'file?');
-var buildCopyUrl = buildUrl.bind(this, 'copy?');
-var buildMoveUrl = buildUrl.bind(this, 'move?');
-var buildMetadataUrl = buildUrl.bind(this, 'data?');
-var buildRevisionsUrl = buildUrl.bind(this, 'revisions?');
-var buildCreateFolderUrl = buildUrl.bind(this, 'folders?');
-
+    var baseUrl = window.contextVars.waterbutlerURL + 'v1/resources/' + nid + '/providers/' + provider + path + suffix;
 
-function buildUploadUrl(path, provider, nid, file, options) {
-    path = (path || '/') + file.name;
-    return buildUrl('file?', path, provider, nid, options);
+    return baseUrl + $.param($.extend(getDefaultOptions(), options));
 }
 
-function buildFromTreebeard(suffix, item, options) {
-    return buildUrl(suffix, item.data.path, item.data.provider, item.data.nodeId, options);
-}
+var buildMetadataUrl = buildUrl.bind(this, '?meta=');
+// buildRevisionsUrl needs to get v1 treatment
+var buildRevisionsUrl = buildUrl.bind(this, '?revisions=');
+var buildCreateFolderUrl = buildUrl.bind(this, '?kind=folder&');
+var buildUploadUrl = buildUrl.bind(this, '?kind=file&');
 
-function buildFromTreebeardFile(item, file, options) {
-    return buildUploadUrl(item.data.path, item.data.provider, item.data.nodeId, file, options);
+function buildFromTreebeard(suffix, item, options) {
+    var path;
+    if ('attributes' in item.data) {
+        path = item.data.attributes.path;
+    } else {
+        path = item.data.path;
+    }
+    return buildUrl(suffix, path, item.data.provider, item.data.nodeId, options);
```

Comment:
If `path` can be in either `item.data` or `item.data.attributes`, can't `provider` be in both, too?  Or is that something Fangorn/Treebeard adds explicitly to `item`.